docs: how to write storybook stories (#6578)
SAP/ui5-webcomponents: docs/6-contributing/04-writing-stories.md

```ts
export const Template = (args) => html`
  <ui5-dialog id="myDialog">
	${unsafeHTML(args.default)}
  </ui5-dialog>
`;

BasicDialog.decorators = [
  (story) => html`
	<ui5-button id="dialogOpener">Open Dialog</ui5-button>
	${story()}
	<script>
		document.getElementById("dialogOpener").addEventListener("click", () => {
			document.getElementById("myDialog").open();
		});
	</script>
  `,
];
```

### Story parameters
[Parameters](https://storybook.js.org/docs/web-components/writing-stories/parameters) are static, metadata about a story, used to control the behavior of Storybook features and addons. 

They can be used both on the default export (Component parameters), applicable to all stories of the component:

```ts
export default {
  title: "main/CheckBox",
  component: "ui5-checkbox",
  parameters: {
	  controls: { include: ["indeterminate", "checked"] }
  }
};
```

Or per every named export (Story Parameters), applicable to a specific story:

```ts
BasicCheckBox.parameters = {
    controls: {
        include: ["indeterminate", "checked"],
    },
};
```

Or globally in the `.storybook/preview.ts` file.

> The above example will include only the `indeterminate`, `checked` properties in the playground for users to edit.

Reviewer: ```suggestion
The above example includes only the `indeterminate`, `checked` properties in the playground for users to edit.
```

## Documentation
The documentation for each component is automatically produced using the `custom-elements.json` file. Additionally, there is an `argTypes.ts` file located beside each `.stories.ts` file. It is generated during build time and contains extra properties that enhance the documentation beyond what is available in the `custom-elements.json` file. This file should not be edited directly, as it can only be modified by the `packages/playground/build-scripts-storybook/samples-prepare.js` script.

### Docs page
Every story has a `docs` page in the storybook's sidebar. Usually, this page is generated automatically by storybook but it can be customized by adding a `docs` property to the story parameters. 

```ts
export default {
    title: "Main/Button",
    component,
    parameters: {
        docs: {
          page: DocsPage({ ...componentInfo, component })
        },
    },
    argTypes,
} as Meta<Button>;
```

## Configuration
Storybook configuration is located in the `.storybook` folder. 

The `preview.ts` file is the main configuration file. It is used to configure the storybook app in terms of UI features, global parameters and decorators. 

> The `main.js` file is used to configure the storybook app in terms of the stories that should be loaded as well as the addons that should be used. Build time configuration is also done here.

Reviewer: ```suggestion
The `main.js` file is used to configure the storybook app in terms of the stories that should be loaded, as well as the addons that should be used. Build time configuration is also done here.
```

> The Manager configured in `manager.tsx` file is the UI responsible for rendering the the search, the toolbar, registering custom addons as well as theming the UI.

Reviewer: ```suggestion
The Manager configured in `manager.tsx` file is the UI responsible for rendering the search, the toolbar, registering custom addons, as well as theming the UI.
```

### Build scripts

> Several build scripts are used to generate the stories documentation. They are located in the `packages/playground/build-scripts-storybook` folder. Ran before every build of the storybook, they are responsible for generating the `argTypes.ts` file, to merge the `custom-elements.json` files and parse the content of the `docs` folder to mdx format.

Reviewer: ```suggestion
Several build scripts are used to generate the stories documentation. They are located in the `packages/playground/build-scripts-storybook` folder. Ran before every build of the storybook, they are responsible for generating the `argTypes.ts` file, to merge the `custom-elements.json` files and parse the content of the `docs` folder to `mdx` format.
```
